### PR TITLE
ui, server: move the node diagnostics page to the admin ui

### PR DESCRIPTION
**Pull request by BramGruneir**
This also pull out the node filtering functions into the nodeFileList.

The ip of each node is now missing, since there is no way to look them up from the ui directly. I'll add it back in, in a follow up PR.
Also, it is not possible set the filter to "node_ids=local". Again, this will be added in a follow up PR.

Replaces #16475.

### CODE REVIEW

**Comment by BramGruneir** [comment]
cc @Amruta-Ranade

**Comment by couchand** [comment]
For the most part looks good, just a few minor comments.

---

Reviewed 4 of 12 files at r1.
Review status: 4 of 12 files reviewed at latest revision, 9 unresolved discussions, some commit checks pending.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 12 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVt0eEu8fjmg9BMi-3:-KpVt0eEu8fjmg9BMi-4:bw485s8) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L12)):*
> ```XML
> import { NodesSummary, nodesSummarySelector } from "src/redux/nodes";
> import * as protos from "src/js/protos";
> import { AdminUIState } from "src/redux/state";
> ```

Could you sort the imports within each group?

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVw5qDVmzyk0Qv3bX8:-KpVw5qDVmzyk0Qv3bX9:b-cwodtx) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L27)):*
> ```XML
>   variable: string;
>   display: string;
>   equality: boolean;
> ```

Even reading the code below it's not entirely clear what this field is used for.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 99 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVuKvMGucgrRZyThuS:-KpVuHxJUGmZo56IndsI:b-tagxww) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L99)):*
> ```XML
> 
>     let nodeIDsContext = _.chain(nodesSummary.nodeIDs)
>       .map(nodeID => Number.parseInt(nodeID, 0));
> ```

s/0/10/

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 105 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVurIT9Wbmfxacvaol:-KpVurIT9Wbmfxacvaom:b-mg55zj) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L105)):*
> ```XML
>     if (!_.isNil(filters.localityRegex)) {
>       nodeIDsContext = nodeIDsContext.filter(nodeID => (
>         !filters.localityRegex.test(localityToString(nodesSummary.nodeStatusByID[nodeID].desc.locality))
> ```

This seems like kind of an awkward way to filter on the locality, particularly since the `localityToString` helper is shared with the other file.  It's probably fine for now, though.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 112 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVvEbNhT_JT3jWlgdR:-KpVvEbNhT_JT3jWlgdS:bnosw1i) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L112)):*
> ```XML
>       .map(nodeID => nodesSummary.nodeStatusByID[nodeID])
>       .map(status => {
>         return {
> ```

It would be nice to integrate these property maps into the `displayList` above as well.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 135 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVxPRfOGzJEhTIfLuo:-KpVxPRfOGzJEhTIfLup:b-45ypaj) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L135)):*
> ```XML
>       .value();
> 
>     function displayResults() {
> ```

It would probably be clearer if this were defined as a method rather than an inner function, and it's conventional to name it something starting with `render`.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 145 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwWrj9kltH_jFLwW2:-KpVwWrj9kltH_jFLwW3:b6hepxa) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L145)):*
> ```XML
>             <tbody>
>               {
>                 _.map(displayList, (title, i) => {
> ```

It may be wise to transpose this table.  The number of fields is bounded but the number of nodes is unbounded.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 162 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwnuBimOfW0OQ_4PT:-KpVwnuBimOfW0OQ_4PU:bs8zpsy) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L162)):*
> ```XML
>                         _.map(nodeDetails, (detail, j) => (
>                           <td key={j} className="nodes-table__cell" title={
>                             _.join(detail[title.variable], "\n")
> ```

I'm not sure this tooltip adds anything useful?

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 166 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVx47TrgEvHyeSQwoC:-KpVx47TrgEvHyeSQwoD:b-ptm960) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L166)):*
> ```XML
>                             <ul className="nodes-entries-list">
>                               {
>                                 _.map(detail[title.variable], (value, k) => (
> ```

There are a lot of levels of looping/markup indentation here, it might be nice to break out one or two of the map functions as auxiliary render methods.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-:-KpVy0VWCnJRzYcY8EC3:bxzoggc)*

**Comment by BramGruneir** [comment]
Thanks for the review!  All comments addressed.

---

Review status: 1 of 12 files reviewed at latest revision, 9 unresolved discussions.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 12 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVt0eEu8fjmg9BMi-3:-KpY4CmcugKuputdW8xB:b-jp7k80) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L12)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Could you sort the imports within each group?
</blockquote></details>

I did.  By final file name.  But I think I should have gone off of full relative directory instead.  fixed.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVw5qDVmzyk0Qv3bX8:-KpY4qDBpPEx0hPRYhTW:b-lgojxx) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L27)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Even reading the code below it's not entirely clear what this field is used for.
</blockquote></details>

Clearly this needed a comment then.  Added.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 99 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVuKvMGucgrRZyThuS:-KpY5Slljz6XD4wyNCWk:b2xwhwc) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L99)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

s/0/10/
</blockquote></details>

hmm... somewhere I had read that 0 would parse it based on a best guess.  I can't find it now.  Done.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 105 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVurIT9Wbmfxacvaol:-KpY5y3J2aoOrHqJXo51:b-b51ubl) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L105)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

This seems like kind of an awkward way to filter on the locality, particularly since the `localityToString` helper is shared with the other file.  It's probably fine for now, though.
</blockquote></details>

The idea here, is that you can pass in a regex for a collection of localities.  So if you have 6 nodes, 
* a=1,b=1
* a=1,b=2
* a=1,b=3
* a=2,b=1
* a=2,b=2
* a=2,b=3

you can pass in `a=1` or `b=2` and get different subsets quickly.  

But you're right that locality to string should be in the component instead of repeated.  I've moved it.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 112 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVvEbNhT_JT3jWlgdR:-KpY93rlJr50lrCS2iIz:b-1z0du7) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L112)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

It would be nice to integrate these property maps into the `displayList` above as well.
</blockquote></details>

What do you mean?  Include a Detail[] within each element in displayList?

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 135 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVxPRfOGzJEhTIfLuo:-KpYAX5fGZM22PxWQI37:b-3u4azs) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L135)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

It would probably be clearer if this were defined as a method rather than an inner function, and it's conventional to name it something starting with `render`.
</blockquote></details>

Done.  Not really sure if it's much clearer.  

What I'm trying to achieve here is to just only display results if there are any.  And the options proposed in https://facebook.github.io/react/docs/conditional-rendering.html all seem to be fairly ugly.  JSX is pretty clear normally, but I find straight conditionals to be kinda ugly.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 145 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwWrj9kltH_jFLwW2:-KpYBIQoq3RmeFM7N3lh:b7tzcy9) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L145)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

It may be wise to transpose this table.  The number of fields is bounded but the number of nodes is unbounded.
</blockquote></details>

In terms of CSS or in terms of how it's displayed?  For display purposes, I think it should remain as is.  I was looking for the right way to do it with CSS and all the solutions seemed kinda hacky.  But I'd be down if you can point me in the right direction.

I did change the left header to be a `<th>` for correctness.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 162 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwnuBimOfW0OQ_4PT:-KpYCjtYHAr6FvSRkEKV:b29cpwi) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L162)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I'm not sure this tooltip adds anything useful?
</blockquote></details>

They're for overflow.  

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 166 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVx47TrgEvHyeSQwoC:-KpYEqCA0zNjuStGE2zy:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L166)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

There are a lot of levels of looping/markup indentation here, it might be nice to break out one or two of the map functions as auxiliary render methods.
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-:-KpYGgtRkgdW1y4-hbhU:b-kdzc89)*

**Comment by couchand** [comment]
Reviewed 3 of 6 files at r2.
Review status: 4 of 12 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 27 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVw5qDVmzyk0Qv3bX8:-Kppd6dyLtlfV9GK-A1D:b-nqzu0z) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L27)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Clearly this needed a comment then.  Added.
</blockquote></details>

That makes it clearer, thanks.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 112 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVvEbNhT_JT3jWlgdR:-KppdWUCLFZYzguF8WDt:b-kuysn2) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L112)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

What do you mean?  Include a Detail[] within each element in displayList?
</blockquote></details>

I mean, each property is mapped from the `status` to the target structure.  Each element of the `displayList` could specify how to map the `status` to extract the results for that `Detail`.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 135 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVxPRfOGzJEhTIfLuo:-KppeD6QS7iN1U-QUKuO:bnpkfnt) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L135)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Done.  Not really sure if it's much clearer.  

What I'm trying to achieve here is to just only display results if there are any.  And the options proposed in https://facebook.github.io/react/docs/conditional-rendering.html all seem to be fairly ugly.  JSX is pretty clear normally, but I find straight conditionals to be kinda ugly.
</blockquote></details>

I see what you mean.  Perhaps the conditional part of it should actually be inline below, and only call into `renderResults` if there are actually details to display.  Something like `_.isEmpty(nodeDetails) ? noNodes : this.renderDetails(nodeDetails)`

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 145 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwWrj9kltH_jFLwW2:-KppeQTLjJYraeu33GcV:b9vdll5) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L145)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

In terms of CSS or in terms of how it's displayed?  For display purposes, I think it should remain as is.  I was looking for the right way to do it with CSS and all the solutions seemed kinda hacky.  But I'd be down if you can point me in the right direction.

I did change the left header to be a `<th>` for correctness.
</blockquote></details>

For display.  As it is will look better as long as there are only a handful of nodes, but for more than that it will start to get out of hand, but OTOH there would be a lot of columns.  Perhaps there's a better way to organize this information so that this wouldn't be a problem.  It seems like there's a lot of unrelated information being thrown into one table here.

I think there's a way to do it with flexbox but I'm not sure if we want to go down that route here...

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 162 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwnuBimOfW0OQ_4PT:-Kppf20PLPPHT8YVN35X:b3q3p2w) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L162)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

They're for overflow.  
</blockquote></details>

Ugh.  Ideally the details wouldn't have to overflow the table, which just goes to my comments above about reorganizing the information.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120)*

**Comment by BramGruneir** [comment]
Review status: 3 of 12 files reviewed at latest revision, 5 unresolved discussions.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 112 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVvEbNhT_JT3jWlgdR:-KpuRu0tLCGeQjIsoK72:bw5ilz5) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L112)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I mean, each property is mapped from the `status` to the target structure.  Each element of the `displayList` could specify how to map the `status` to extract the results for that `Detail`.
</blockquote></details>

I tried doing this.  But I found that it just became so abstract that it was quite hard to follow.  Going to leave it as is.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 135 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVxPRfOGzJEhTIfLuo:-KpuT-PKsCMtsQe7Uah-:b96g8wp) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L135)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I see what you mean.  Perhaps the conditional part of it should actually be inline below, and only call into `renderResults` if there are actually details to display.  Something like `_.isEmpty(nodeDetails) ? noNodes : this.renderDetails(nodeDetails)`
</blockquote></details>

I found a cleaner solution.  I put the early exit for no results in a return before it gets to the main render method.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 145 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwWrj9kltH_jFLwW2:-KpuS0WYdVl4l7yuTAf9:b-g2hsgw) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L145)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

For display.  As it is will look better as long as there are only a handful of nodes, but for more than that it will start to get out of hand, but OTOH there would be a lot of columns.  Perhaps there's a better way to organize this information so that this wouldn't be a problem.  It seems like there's a lot of unrelated information being thrown into one table here.

I think there's a way to do it with flexbox but I'm not sure if we want to go down that route here...
</blockquote></details>

Since this is port from the golang templates to the admin ui, I'm going to keep the format as close to the original as possible.  In the 1.2 timeframe, I think we're going to be doing a full redesign of these pages.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 162 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwnuBimOfW0OQ_4PT:-KpuS6B6NXHghf70dely:bxxst7y) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L162)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Ugh.  Ideally the details wouldn't have to overflow the table, which just goes to my comments above about reorganizing the information.
</blockquote></details>

I agree.  But until we redesign the pages, I'm going to leave this as is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120)*

**Comment by BramGruneir** [comment]
I think is good to merge.  Any last comments?

---

Review status: 3 of 12 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-:-KpuhQuue4P2fXZEkLjB:bbtbl6h)*

**Comment by couchand** [comment]
:lgtm:

---

Reviewed 1 of 4 files at r3.
Review status: 4 of 12 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 112 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVvEbNhT_JT3jWlgdR:-Kpuyx6rpOyIqYTCtm-L:b-gxztsp) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L112)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

I tried doing this.  But I found that it just became so abstract that it was quite hard to follow.  Going to leave it as is.
</blockquote></details>

Fair enough.

---

*[pkg/ui/src/views/reports/containers/nodes/index.tsx, line 145 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-KpVwWrj9kltH_jFLwW2:-Kpuz-KQnCo4Xqex6iMZ:b-32w0iv) ([raw file](https://github.com/cockroachdb/cockroach/blob/94be24af484fd2f54ba0a19062fb276ecc06da0b/pkg/ui/src/views/reports/containers/nodes/index.tsx#L145)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Since this is port from the golang templates to the admin ui, I'm going to keep the format as close to the original as possible.  In the 1.2 timeframe, I think we're going to be doing a full redesign of these pages.
</blockquote></details>

Sounds reasonable.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17120#-:-KpuzH_TeeJ7cUaYkO6e:bnfp4nl)*
